src/collective/fingerpointing/browser/logview.py

```diff
@@ -32,10 +33,28 @@ def tail(self, f, n=100):
         return lines[-n:]
 
     @property
+    def _get_audit_log_files(self):
+        """Return a list with the current audit log and all of its
+        backups. The list is sorted with newer files first.
+        """
+        # retrieve the list of audit log backup files
+        logfiles = sorted([
+            f for f in glob(LOGFILE + '.*') if '.lock' not in f
+        ], reverse=True)
+        # include active audit log as first element
+        logfiles.insert(0, LOGFILE)
+        return logfiles
+
+    @property
     def get_audit_log(self):
         """Return audit log with newer entries first."""
-        with open(logfile, 'r') as fp:
-            lines = self.tail(fp)
+        lines = []
+        # process all logs until we have enough lines
+        for logfile in self._get_audit_log_files:
+            with open(logfile, 'r') as fp:
+                lines.extend(self.tail(fp))
+            if len(lines) >= 100:
+                break
 
-        lines.reverse()
+        lines = sorted(lines[:100], reverse=True)
```

Comment:
current implementation already takes into account handling of large files in an efficient way